compiler/noirc_evaluator/src/ssa/opt/basic_conditional.rs

```diff
@@ -0,0 +1,522 @@
+
+    // 1. Process all blocks of the cfg, starting from the root and following the successors
+    while let Some(block) = stack.pop() {
+        // Avoid cycles
+        if processed.contains(&block) {
+            continue;
+        }
+        processed.insert(block);
+
+        // Identify the simple conditionals
+        if let Some(conditional) = is_conditional(block, &cfg, function) {
+            // no need to check the branches, process the join block directly
+            stack.push(conditional.block_exit);
+            conditionals.push(conditional);
+        } else {
+            stack.extend(cfg.successors(block));
+        }
+    }
+
+    // 2. Flatten all simple conditionals
+    // process basic conditionals in reverse order so that
+    // a conditional does not impact the previous ones
+    conditionals.reverse();
+    Context::flatten_multiple(&conditionals, function, no_predicates);
+}
+
+impl<'f> Context<'f> {
+    fn flatten_single_conditional(
+        &mut self,
+        conditional: &BasicConditional,
+        no_predicates: &mut HashMap<FunctionId, bool>,
+    ) {
+        // Manually inline 'then', 'else' and 'exit' into the entry block
+        //0. initialize the context for flattening a 'single conditional'
+        let mut queue = vec![];
+        self.target_block = conditional.block_entry;
+        self.no_predicate = true;
+        //1. process 'then' branch
+        self.inline_block(conditional.block_entry, no_predicates);
+        let to_process = self.handle_terminator(conditional.block_entry, &queue);
+        queue.extend(to_process);
+        if let Some(then) = conditional.block_then {
+            assert_eq!(queue.pop(), conditional.block_then);
+            self.inline_block(then, no_predicates);
+            let to_process = self.handle_terminator(then, &queue);
+
+            for incoming_block in to_process {
+                if !queue.contains(&incoming_block) {
+                    queue.push(incoming_block);
+                }
+            }
+        }
+
+        //2. process 'else' branch, in case there is no 'then'
+        let next = queue.pop();
+        if next == conditional.block_else {
+            let next = next.unwrap();
+            self.inline_block(next, no_predicates);
+            let to_process = self.handle_terminator(next, &queue);
+            for incoming_block in to_process {
+                if !queue.contains(&incoming_block) {
+                    queue.push(incoming_block);
+                }
+            }
+        } else {
+            assert_eq!(next, Some(conditional.block_exit));
+        }
+
+        //3. process 'exit' block
+        self.inline_block(conditional.block_exit, no_predicates);
+        // Manually set the terminator of the entry block to the one of the exit block
+        let terminator =
+            self.inserter.function.dfg[conditional.block_exit].terminator().unwrap().clone();
+        let mut next_blocks = VecDeque::new();
```

Comment:
I may be blind, but where is this consumed?

Reply:
Good point, this part is copy/paste from the flatten_cfg pass but slightly adapted, hence the comment 'manually inline'. The 'next blocks to process' are not needed here because there is no nested if nor loops.

Reply:
Right, I didn't know it was based on an existing pass, but then can we remove it from here?